src/ShadowDOM/wrappers/Document.js

```diff
@@ -338,6 +338,13 @@
     'hasFeature',
   ]);
 
+  var originalCreateTreeWalker = document.createTreeWalker;
+  function createTreeWalker(root, whatToShow, filter, entityReferenceExpansion){
+    return originalCreateTreeWalker.call(document,unwrap(root), whatToShow, filter, entityReferenceExpansion);
```

Comment:
From https://developer.mozilla.org/en-US/docs/Web/API/Document/createTreeWalker
it looks like the filter should be wrapped, something like:
```js
// Make sure we only pass wrapped nodes to the user's filter method
var newFilter = function(node) { return filter(wrap(node)); }
```

... and then the tree walker API's need to be wrapped too:
https://developer.mozilla.org/en-US/docs/Web/API/TreeWalker

those methods should all return wrapped nodes. Something like this:

```js
// wrappers/TreeWalker.js
function TreeWalker(impl) {
  setWrapper(this, impl);
}
mixin(TreeWalker.prototype, {
  nextNode: function() { return wrapIfNeeded(unsafeUnwrap(this).nextNode()); },
  ...
});

registerWrapper(...)
```

here's an example of the pattern: https://github.com/webcomponents/webcomponentsjs/blob/master/src/ShadowDOM/wrappers/Selection.js